iquip/apps/dataviewer.py

```diff
@@ -165,6 +197,22 @@ def setData(self, data: np.ndarray, axes: Sequence[AxisInfo]):
         width, height = haxis.values[-1] - x, vaxis.values[-1] - y
         self.image.setRect(x, y, width, height)
 
+    def nearestDataPoint(self,
+        scenePos: pg.Point,
+        tolerance: Optional[float] = None,  # pylint: disable=unused-argument
+    ) -> Optional[Tuple[int, int]]:
+        """Overridden.
+        
+        ImageViewer does not use tolerance since it has a clear bounding box for
+          each data point.
+        """
+        dataPos = self.image.mapFromDevice(scenePos)
+        x, y = np.floor(dataPos.x()), np.floor(dataPos.y())
```

Comment:
Why did you use `np.floor()`, not `int()`?

Reply:
I also used it at the first time, but noticed we forgot that `int()` truncates towards zero. That means -0.5 is mapped to 0, so they are misinterpreted as valid clicks.

Reply:
Wow, I forgot it perfectly! Thanks for noticing me.